src/tcache.c

```diff
@@ -492,20 +492,23 @@ tcache_init(tsd_t *tsd, tcache_t *tcache, void *avail_stack) {
 
 static size_t
 tcache_bin_stack_alignment (size_t size) {
+	if (LG_SIZEOF_PTR < 3 || size < PAGE) {
```

Comment:
Shouldn't this be:
```
size_t correctness_alignment = pow2_ceil_zu(size);
size_t performance_alignment = CACHELINE; // Or page, whatever granularity we pick.
return max(correctness_alignment, performance_alignment);  
}
```
?
I don't follow why pointer size matters here at all.

Reply:
Ah the pointer size is because, the alignment is to avoid overflowing the "half" pointers in the cache bin metadata. Those are 32bits pointers so no overflow concerns there.

Reply:
Added some comments for the 32bit case.